### PR TITLE
Feature/max header size

### LINKED ISSUE #5: Sanity checks - Max header size
As specified by Undertow

.setServerOption(UndertowOptions.MAX_HEADER_SIZE, 2048)

### PR DESCRIPTION
**Pull request by zhaoyuango**
# Description
This PR add support for validating max header size.

Fixes #5 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?
see unit test.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/test/java/io/vacco/mrx/MxConfigTest.java [hunk 1]
```diff
@@ -44,6 +44,13 @@ public class MxConfigTest {
         var res3 = client.send(req, ofString());
```
Comment (jjzazuet) [inline src/test/java/io/vacco/mrx/MxConfigTest.java:44]: @zhaoyuango can you rename this variable to `res`? Thanks!

**Review by jjzazuet — CHANGES_REQUESTED** [review]
Looks good overall. Just make a few variable cleanup changes and we should be good. Thanks!

**Comment by zhaoyuango** [comment]
thanks

File: src/test/java/io/vacco/mrx/MxConfigTest.java (hunk 1 above)
Comment (zhaoyuango) [reply src/test/java/io/vacco/mrx/MxConfigTest.java]: thanks

**Review by jjzazuet — APPROVED** [review]
LGTM

**Pull request merged**